File: Data Structures & Algorithms/minimum-difference-between-highest-and-lowest-of-k-scores/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number}
     */
    minimumDifference(nums, k) {
        nums.sort((a,b) => a - b)
        let l = 0,
            r = k - 1,
            res = Infinity;
        while ( r < nums.length){
            res = Math.min(res, nums[r] - nums[l])
            r++;
            l++
        }
        return res
    }
}

File: Data Structures & Algorithms/max-consecutive-ones-iii/submission-0.cpp
class Solution {
public:
    int longestOnes(vector<int>& nums, int k) {
        int res = 0;
        int curr = 0;
        int currFlip = 0;
        int l = 0, r = 0;
        while(r < nums.size()){
            if(nums[r] == 1){
                curr++;
                r++;
            }else if(nums[r] == 0 && currFlip < k){
                curr++;
                currFlip++;
                r++;
            }
            else{
                if(nums[l] == 0){
                    currFlip--;
                }
                l++;
                curr--;
            }
            res = max(curr, res);
        }
        return res;
    }
};